recipes/osgearth/all/conanfile.py

```diff
@@ -1,206 +1,250 @@
+        tc.variables["OSGEARTH_ASSUME_SINGLE_THREADED_OSG"] = self.options.assume_single_threaded_osg
+        tc.variables["OSGEARTH_INSTALL_SHADERS"] = self.options.install_shaders
+        tc.generate()
+
+        deps = CMakeDeps(self)
+        # Use the same name for shared and static targets
+        deps.set_property("rocksdb", "cmake_target_aliases", ["RocksDB::rocksdb", "RocksDB::rocksdb-shared"])
+        deps.generate()
+
+        VirtualBuildEnv(self).generate()
 
-        self._patch_sources()
+    def _patch_sources(self):
+        apply_conandata_patches(self)
 
-    @functools.lru_cache(1)
-    def _configured_cmake(self):
+    def build(self):
+        self._patch_sources()
         cmake = CMake(self)
-        cmake.definitions["OSGEARTH_BUILD_SHARED_LIBS"] = self.options.shared
-        cmake.definitions["OSGEARTH_BUILD_TOOLS"] = False
-        cmake.definitions["OSGEARTH_BUILD_EXAMPLES"] = False
-        cmake.definitions["OSGEARTH_BUILD_TESTS"] = False
-
-        cmake.definitions["OSGEARTH_BUILD_PROCEDURAL_NODEKIT"] = self.options.build_procedural_nodekit
-        # cmake.definitions["OSGEARTH_BUILD_TRITON_NODEKIT"] = self.options.build_triton_nodekit
-        # cmake.definitions["OSGEARTH_BUILD_SILVERLINING_NODEKIT"] = self.options.build_silverlining_nodekit
-        cmake.definitions["OSGEARTH_BUILD_LEVELDB_CACHE"] = self.options.build_leveldb_cache
-        cmake.definitions["OSGEARTH_BUILD_ROCKSDB_CACHE"] = self.options.build_rocksdb_cache
-        cmake.definitions["OSGEARTH_BUILD_ZIP_PLUGIN"] = self.options.build_zip_plugin
-        cmake.definitions["OSGEARTH_ENABLE_GEOCODER"] = self.options.enable_geocoder
-
-        cmake.definitions["WITH_EXTERNAL_DUKTAPE"] = False
-        cmake.definitions["WITH_EXTERNAL_TINYXML"] = False
-        cmake.definitions["CURL_IS_STATIC"] = not self.options["libcurl"].shared
-        cmake.definitions["CURL_INCLUDE_DIR"] = self.deps_cpp_info["libcurl"].include_paths[0]
-        cmake.definitions["OSGEARTH_INSTALL_SHADERS"] = self.options.install_shaders
-        cmake.definitions["OSGEARTH_ENABLE_NVTT_CPU_MIPMAPS"] = self.options.enable_nvtt_cpu_mipmaps
-        cmake.definitions["OSGEARTH_ENABLE_WININET_FOR_HTTP"] = self.options.enable_wininet_for_http
-
-        # our own defines for using in our top-level CMakeLists.txt
-        cmake.definitions["OSGEARTH_WITH_GEOS"] = self.options.with_geos
-        cmake.definitions["OSGEARTH_WITH_SQLITE3"] = self.options.with_sqlite3
-        cmake.definitions["OSGEARTH_WITH_WEBP"] = self.options.with_webp
-
         cmake.configure()
-
-        return cmake
-
-    def build(self):
-        self._configured_cmake().build()
+        cmake.build()
 
     def package(self):
-        self._configured_cmake().install()
-        self.copy(pattern="LICENSE.txt", dst="licenses", src=self._source_subfolder)
-
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
```

Comment:
Correct license filename needs to be retained:

```suggestion
        copy(self, "LICENSE.txt", self.source_folder, os.path.join(self.package_folder, "licenses"))
```